src/dnmd/tables.c

```diff
@@ -268,16 +276,18 @@ bool initialize_table_details(
     if (all_table_row_counts[id] == 0)
         return false;
 
-    mdtcol_t const string_index = mdtc_idx_heap | mdtc_hstring | (heap_sizes & 0x1 ? mdtc_b4 : mdtc_b2);
-    mdtcol_t const guid_index = mdtc_idx_heap | mdtc_hguid | (heap_sizes & 0x2 ? mdtc_b4 : mdtc_b2);
-    mdtcol_t const blob_index = mdtc_idx_heap | mdtc_hblob | (heap_sizes & 0x4 ? mdtc_b4 : mdtc_b2);
+    mdtcol_t const string_index = mdtc_idx_heap | mdtc_hstring | (context_flags & 0x1 ? mdtc_b4 : mdtc_b2);
+    mdtcol_t const guid_index = mdtc_idx_heap | mdtc_hguid | (context_flags & 0x2 ? mdtc_b4 : mdtc_b2);
+    mdtcol_t const blob_index = mdtc_idx_heap | mdtc_hblob | (context_flags & 0x4 ? mdtc_b4 : mdtc_b2);
```

Comment:
Probably want to update the checked bit masks to the enum values (i.e., `mdcxt_flag_t`).